_posts/en/newsletters/2022-07-06-newsletter.md

```diff
@@ -0,0 +1,186 @@
+  LN forwarding nodes can use in collecting fees for routing payments
+  (including the strategy of not collecting fees).  ZmnSCPxj then
+  analyzes possible outcomes of the different strategies.  This seems
+  related to his proposal for nodes to use routing fees to improve
+  payment success rates, see [Newsletter #204][news204 fee signal],
+  which also received significant [additional commentary][towns fee
+  signal] in the past week from Anthony Towns.
+
+- **Onion message rate limiting:** Bastian Teinturier
+  [posted][teinturier rate limit] a summary of an idea he attributes to
+  Rusty Russell for rate limiting [onion messages][topic onion
+  messages].  The proposal has each node store just an extra 32 bytes of
+  information for each of their peers that allows them to
+  probabalistically punish peers who send too much traffic.  The
+  suggested penalty is to just half the rate limit for a peer relaying
+  too much traffic for about 30 seconds.  It's acceptable if this
+  lightweight penalty is occasionally enforced against the wrong peer,
+  as may happen with this idea.  The proposal also allows the originator
+  of a message to learn which downstream nodes are rate limiting their
+  messages (again probabalistically), giving them a chance to resend the
+  message by a different route.
+
+    Olaoluwa Osuntokun [suggested][osuntokun onion pay] reconsideration
+    of his previous proposal for preventing abuse of onion messages by
+    charging for data relay, see [Newsletter #190][news190 onion pay].
+    Replies from other developers as of this writing seemed to indicate
+    they first try the lightweight rate limiting to see if it works
+    before adding the complexity of payments for onion messages.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+FIXME:harding to update Tuesday
+
+- LDK
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #24836][] add RPC (-regtest only) for testing package policy FIXME:glozow
+
+- [Bitcoin Core #22558][] psbt: Taproot fields for PSBT FIXME:bitschmidty
+
+- [Core Lightning #5281][] adds support for specifying the `log-file`
+  configuration option multiple times to write to multiple log files.
+
+- [LDK #1555][] updates its pathfinding code to slightly prefer routing
+  through channels which advertise that they won't accept payments
+  larger than half the amount of money committed to the channel.  This
```

Comment:
I first understood this to mean that the channel will limit to a total of half being in pending forwards in either direction. However, looking at the code, it seems to me that LDK discourages channels that allow a single HTLC to use _half or more_ of the capacity. (Unless `htlc_maximum_msat` pertains to the total rather than a single HTLC, which [doesn't seem to be the case](https://github.com/lnbook/lnbook/blob/ece69d5c2ac8116ef83c1826bd43bd4b33c74dca/10_onion_routing.asciidoc).)

To achieve the privacy benefit you mention, it seems to me that instead `max_htlc_value_in_flight_msat` would have to be restricted to less than half in both directions, so I'm a bit confused. Otherwise they could just create multiple HTLCs? :thinking:

Reply:
From my understanding:

- `max_htlc_value_in_flight_msat` (defined in BOLT2) is a peer-level attribute that allows Alice to tell her channel partner Bob that she won't accept more then *n* msats aggregated across all pending HTLCs.
- `htlc_maximum_msat` (defined in BOLT7) is a gossiped channel attribute that must be set to less than or equal to `max_htlc_value_in_flight_msat` and is (as you say) that max per HTLC

A node that's pathfinding based on public gossip won't know the `in_flight_msat` for any channels besides its direct peers, so it can't actually use that value directly---but it may be able to use `htlc_maximum_msat` as a reasonable proxy.  It certainly seems *possible* that someone could set `in_flight_msat` to ~100% of channel capacity but `htlc_maximum_msat` to 50% of capacity, but it perhaps seems more likely to LN devs that any node limiting `htlc_maximum_msat` to 50% or less of capacity has also limited `in_flight_msat` to 50% or less.

Does that make sense?  If not, I suggest we drop this description from this week's newsletter until I have a chance to talk to an LDK dev for clarification.

Reply:
Also, thanks so much for examining this in detail!

Reply:
My pleasure. Just to clarify in case that got dropped under the table, the 50% is part of the penalized range. :)

I could ping an LDK dev to take a look. Lemme see if I can get a hold of someone.

Reply:
The observation that `max_htlc_value_in_flight_msat` is not a gossiped value certainly makes evaluating `htlc_maximum_msat` more relatable. One of the involved devs said they'll be able to take a look later today.

Reply:
Updated, thanks!